sdk/cosmosdb/cosmos/src/extractPartitionKey.ts

```diff
@@ -33,8 +35,9 @@ export function extractPartitionKey(
     if (partitionKey.length === 1 && partitionKey[0] === undefined) {
       return undefinedPartitionKey(partitionKeyDefinition);
     }
-    return partitionKey;
   }
+
+  return partitionKey;
```

Comment:
I think the return type has been wrong before this PR, but this seems to return `undefined` if the above if statement returns false
```
partitionKeyDefinition &&
partitionKeyDefinition.paths &&
partitionKeyDefinition.paths.length > 0
```